### ISSUE TITLE
Pouch DB vs trailing slash

### ISSUE THREAD

**Issue by andrew-aladev** [pouchdb/pouchdb#5485]
I've found a place where developers does not care about trailing slash.
```
// Generate a URL with the host data given by opts and the given path
function genDBUrl(opts, path) {
  return genUrl(opts, opts.db + '/' + path);
}
```
`var dbUrl = genDBUrl(host, '');`
`var checkExists = {method: 'GET', url: dbUrl};`

This will be a problem for anyone who wants to use strict url for db. Thank you.

**Comment by nolanlawson** [comment]
Sorry, I'm not quite understanding this bug report. Do you mean to say that if you serve a website at `http://example.com/foo` and you also want the CouchDB to live at `http://example.com/foo` then you can't pass in `new PouchDB('')`? Or are you saying something else?

**Comment by andrew-aladev** [comment]
Hello. The code is simple:

```
new PouchDB('http://localhost:3000/test').put({ "_id" : "1" });
```

If debug is enabled you will see:
```
pouchdb:http GET http://localhost:3000/test/
pouchdb:http POST http://localhost:3001/test/_bulk_docs
```

This is cased by:

```
function setup() {
...
var checkExists = {method: 'GET', url: dbUrl};
setupPromise = ajaxPromise({}, checkExists)...
```

I was expecting a query to "/test/_bulk_docs". The query to "/test/" made me confused. The program that I am working with was confused too. This url doesn't look good.

I think trailing slash in this url can be stripped without harm to setup function. I don't know whether this is the only place where trailing slash can be stripped. Thank you.

**Comment by daleharvey** [comment]
So CouchDB or any server shouldnt care whether the trailing slash is there, its entirely aesthetic, I expect a trailing slash mostly because there are things under it, is this causing any problems? does anyone want to give reasons for / against having the trailing slash?

**Comment by nolanlawson** [comment]
It shouldn't matter at all assuming the proxy between the client and CouchDB is set up correctly. CouchDB resolves the trailing slash appropriately.

**Comment by yaronyg** [comment]
Short Answer: I think we can close this bug

Long Answer:

The reason this came up is that we are using an ACL library (that we wrote) that generically enforces ACLS for HTTP requests (not just CouchDB). In the HTTP object model foo and foo/ are two different resources. It is the case that they often (as in CouchDB) mapped to the same underlying resource but that is not guaranteed. So in a generic ACL system for HTTP it really matters if the / is expected or not. It so happened that our config required a DB to not be named with a trailing / so we were failing even though we submitted a name without a trailing slash because PouchDB was adding that trailing slash. But since we anyway want to be consistent with CouchDB's behavior (where the trailing / is always mapped to the same resource without the trailing /) we are going to change our library to support a 'don't worry about trailing slash' mode.

**Comment by daleharvey** [comment]
Yup anything that treats `/foo` and `/foo/` different is going to have problems, if there were particular reasons for a preference either way would be happy to take a patch, but right now it seems like an arbitrary aesthetic choice so closing out, cheers

**Comment by alexfedoseev** [comment]
Sorry for pinging such an old issue but I just bumped into this. In my nginx, I configured 301 redirect `/path/ -> /path` and noticed that in dev-tools Pouchdb does 2 requests: 1st to endpoint w/ trailing slash -> redirect -> endpoint w/o trailing slash. Does it makes sense to have an option in http plugin, something like `trailingSlash: bool`, so it's possible to avoid redirect on each request for servers w/ such configuration?
